Wigle CSV formatting, scan hidden networks

lukeswitz opened the pull request:
### What's Changed
[+] Wigle CSV formatting
[+] Capture authentication type
[+] Scan/display hidden networks
[+] Unify GPS data structure

lukeswitz committed:
Wigle CSV formatting, scan hidden networks

- unify GPS data structure

ZeroChaos- commented on m5paperwardriver/m5paperwardriver.ino:
```diff
@@ -155,26 +178,31 @@ void loop() {
   }
 
-  int n = WiFi.scanNetworks();
+  int n = WiFi.scanNetworks(false,true,false,110); // Scan hidden networks at 110ms per channel
```
most things (such as kismet) do 200ms by default.  110ms is enough, but 200ms would notably increase your chances as you can lose 1 packet (50%) and still see the AP

ZeroChaos- commented on m5paperwardriver/m5paperwardriver.ino:
```diff
@@ -186,3 +214,28 @@ void loop() {
+    case WIFI_AUTH_WAPI_PSK:
+      return "[WAPI_PSK]";
+    default:
+      return "[UNKNOWN]";
```
this function is a great idea, but is it possible to return a string version of wifiAuth in the default case so we can manually figure it out and then more easily add support?

lukeswitz replied:
110 works good. Adapted long ago from
https://github.com/JosephHewitt/wardriver_rev3/blob/main/A/A.ino#L2481

Feel free to use any values you please. Mod CSVs later to add unknowns without slowing it down. Not sure I understand that part. Thanks for the knowledge

ZeroChaos- replied:
In this statement `switch (wifiAuth) {` you read from a variable and pull a bunch of known cases out, which is great.  But in the case of default, you return only `[UNKNOWN]'.  it might be more useful to simply return the value from wifiAuth, or the value from wifiAuth with a known prefix or suffix indicating it's a raw value.  I don't know if you implemented all the current values from the source code, but even if you did it is missing a few which may be added later and returning the raw value would be more useful to the user than a generic "[UNKNOWN]"

lukeswitz replied:
OK.

lukeswitz replied:
> In this statement `switch (wifiAuth) {` you read from a variable and pull a bunch of known cases out, which is great. But in the case of default, you return only `[UNKNOWN]'. it might be more useful to simply return the value from wifiAuth, or the value from wifiAuth with a known prefix or suffix indicating it's a raw value. I don't know if you implemented all the current values from the source code, but even if you did it is missing a few which may be added later and returning the raw value would be more useful to the user than a generic "[UNKNOWN]"

1. They’re written in string format to the CSV. 

2. Yes, because these are what the esp32 WiFi library supports: 

- https://docs.espressif.com/projects/arduino-esp32/en/latest/api/wifi.html

- https://docs.espressif.com/projects/esp-idf/en/stable/esp32/api-reference/network/esp_wifi.html

- https://www.arduino.cc/reference/en/libraries/wifi/wifi.scannetworks/

As for dwell times 200 is the standard, but in tests around 150 was about perfect for many M5 units. Tested this morning with no improvement. 

ChatGPT told me same as it told you “increasing the dwell time to 200ms provides a better opportunity to capture essential packets, as losing even 50% of the transmissions could still allow you to detect the access point​(
[Kismet](https://www.kismetwireless.net/docs/readme/intro/passive_capture/)
).”

To that end I’d recommend opening a PR on the linked wardriver code by Hewitt as it’s deployed worldwide. This is a single device hobby project. 

Regards.

ZeroChaos- replied:
thanks for the suggestion, I opened an issue to discuss the dwell change.  I don't have a ton of experience with the embedded wifi stuff so I wanted feedback in case there was something different which I don't know.
https://github.com/JosephHewitt/wardriver_rev3/issues/177

lukeswitz replied:
Nice work. It’s safer to dwell longer with more interfaces that’s for sure. I run these faster because loss has been negligible/nil. On larger rigs I notice a bunch of issues with interference when hopping faster then 5 ch/s.

lukeswitz committed:
Use Nyquist sampling as ZeroChaos suggested 

Sometimes packets are missed using multiple interfaces, rarely on these M5 esp32s does it happen. Tradeoff...reliability:speed

ZeroChaos- commented:
Not my repo to approve, but this all looks like improvements to me. :-)  Thanks to you both, I bought one of these and I'm excited to play with it.

colonelpanichacks commented:
nice work!
